File: app/src/main/java/com/example/jorgecaro/almacenamiento_jorge_caro/repositories/Local/UserDao.java
package com.example.jorgecaro.almacenamiento_jorge_caro.repositories.Local;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import com.example.jorgecaro.almacenamiento_jorge_caro.model.User;

import java.util.List;

/**
 * Created by jorge caro on 7/27/2017.
 */

public class UserDao implements IDao<User>{

    private static UserDao instance;
    private Database database;

    private UserDao(Context context){
        this.database = new Database(context);
    }

    public static UserDao getInstance(Context context){
        if(instance == null) instance = new UserDao(context);
        return instance;
    }

    @Override
    public boolean add(User user){
        SQLiteDatabase sq = database.getWritableDatabase();
        ContentValues valores = new ContentValues();
        valores.put(Database.getUserFields()[1], user.getNombre());
        valores.put(Database.getUserFields()[2], user.getPass());
        long res = sq.insert(Database.getUserTable(),null, valores);
        return res > 0;
    }

    public boolean checkIfExist(User user){
        SQLiteDatabase sq = database.getWritableDatabase();
        Cursor cursor = sq.rawQuery("SELECT * FROM USER WHERE " +Database.getUserFields()[1]+" = ? AND " +Database.getUserFields()[2]+" = ?"
                                    , new String[]{user.getNombre()
                                    , user.getPass()});
        return cursor.getCount() > 0;
    }

    public boolean update(User user){
        SQLiteDatabase sq = database.getWritableDatabase();
        ContentValues valores = new ContentValues();
        long res = sq.update(Database.getUserTable(),
                            valores,Database.getUserFields()[0]+" = ? "
                            , new String[]{user.getId()+""});
        return res > 0;
    }

    @Override
    public List<User> search(int id) {
        return null;
    }

    @Override
    public boolean delete(User user){
        SQLiteDatabase sq = database.getWritableDatabase();
        return sq.delete(Database.getUserTable(),
                        Database.getUserFields()[1] + "= ?"
                        +" AND "+Database.getUserFields()[2] + "= ?"
                        , new String[]{ user.getNombre(),  user.getPass()}) > 0;
    }

    @Override
    public List<User> search() {
        return null;
    }
}

File: app/src/main/java/com/example/jorgecaro/almacenamiento_jorge_caro/repositories/Local/UserDao.kt
package com.example.jorgecaro.almacenamiento_jorge_caro.repositories.Local

import android.content.ContentValues
import android.content.Context
import com.example.jorgecaro.almacenamiento_jorge_caro.model.User

/**
 * Created by jorge caro on 7/27/2017.
 */

class UserDao(context: Context) : IDao<User> {

    private var database = Database(context)

    override fun add(entity: User): Boolean {
        val sq = database.writableDatabase
        val values = ContentValues()
        values.put(Database.userFields[1], entity.nombre)
        values.put(Database.userFields[2], entity.pass)
        val res = sq.insert(Database.userTable, null, values)
        return res > 0
    }

    override fun checkIfExist(entity: User): Boolean {
        val sq = database.writableDatabase
        val cursor = sq.rawQuery("SELECT * FROM USER WHERE " + Database.userFields[1] + " = ? AND " + Database.userFields[2] + " = ?", arrayOf(entity.nombre, entity.pass))
        return cursor.count > 0
    }

    override fun update(entity: User): Boolean {
        val sq = database.writableDatabase
        val valores = ContentValues()
        val res = sq.update(Database.userTable,
                valores, Database.userFields[0] + " = ? ", arrayOf(entity.id.toString() + "")).toLong()
        return res > 0
    }

    override fun search(id: Int): List<User> {
        TODO()
    }

    override fun delete(entity: User): Boolean {
        val sq = database.writableDatabase
        return sq.delete(Database.userTable,
                Database.userFields[1] + "= ?"
                        + " AND " + Database.userFields[2] + "= ?", arrayOf(entity.nombre, entity.pass)) > 0
    }

    override fun search(): List<User> {
        TODO()
    }

    override fun searchById(id: Int): User {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}
